src/main/scala/nl/knaw/dans/easy/stage/fileitem/FileItemCsv.scala

```diff
@@ -0,0 +1,51 @@
+package nl.knaw.dans.easy.stage.fileitem
+
+import java.io.InputStream
+
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.csv.CSVParser.parse
+import org.rogach.scallop.TrailingArgsOption
+import org.slf4j.LoggerFactory
+
+import scala.collection.JavaConversions._
+import scala.io.Source.fromInputStream
+import scala.util.{Success, Try}
+
+object FileItemCsv {
+  val log = LoggerFactory.getLogger(getClass)
+
+  def read(in: InputStream, conf: FileItemConf): Try[Array[FileItemSettings]] = Try {
+    val trailArgs: List[String] = List(conf.sdoSetDir.apply().toString)
+    val csv = {
+      val rawContent = fromInputStream(in).mkString
+      parse(rawContent, CSVFormat.RFC4180).getRecords.map(_.toList)
+    }
+    if (csv.isEmpty){
+      log.warn("CSV file empty")
+      return Success(Array[FileItemSettings]())
+    }
```

Comment:
Why make this a "success" and log a warning here? Just make it a Failure and log the problem in the caller, or still better make it a fatal error. An empty CSV file isn't very useful after all.